### ISSUE TITLE
[Tracking Issue] COSMIC Desktop Environment 

### ISSUE THREAD

**Issue by Nyabinary** [NixOS/nixpkgs#259641]
**Project description**
COSMIC: Next-generation DE made by System76 (and in Rust)
[ ] cosmic-epoch  
[x] cosmic-applets #251365 
[ ] cosmic-applibrary 
[ ] cosmic-bg 
[x] cosmic-comp #251365 
[ ] cosmic-icons 
[ ] cosmic-launcher 
[ ] cosmic-notifications 
[ ] cosmic-osd 
[x] cosmic-panel #251365 
[ ] cosmic-session
[ ] cosmic-settings
[ ] cosmic-settings-daemon
[ ] cosmic-workspaces-epoch
[ ] xdg-desktop-portal-cosmic
[ ] cosmic-greeter
[ ] cosmic-text
[ ] libcosmic
[ ] cosmic-protocols
[ ] cosmic-time
[ ] cosmic-text-editor
[ ] cosmic-screenshot

* homepage URL: https://github.com/pop-os/
* source URL: https://github.com/pop-os/
* license: gpl3
* platforms: linux

**Comment by Nyabinary** [comment]
> You should add cosmic-greeter as well:
> 
> https://github.com/pop-os/cosmic-greeter

Already added that :)

**Comment by ahoneybun** [comment]
Ah I was looking at the list like it was A-Z and I missed it, sorry!

**Comment by Nyabinary** [comment]
https://blog.system76.com/post/locked-and-loaded-with-new-cosmic-de-updates/
New COSMIC blog, I wonder how feasible a COSMIC NixOS channel is in Matrix considering GNOME has one.

**Comment by viperML** [comment]
Looking forward to a matrix channel to cooperate the effort. I would be intersted in contributing to the packages or module(s).

**Comment by GaetanLepage** [comment]
> Looking forward to a matrix channel to cooperate the effort. I would be intersted in contributing to the packages or module(s).

Same here !

**Comment by Nyabinary** [comment]
@viperML @GaetanLepage I made an *unofficial* Matrix space that *isn't* officially authorized by the Nix Foundation right here:
https://matrix.to/#/#cosmic-nixos:conduit.rs

**Comment by Nyabinary** [comment]
Hmm how else can we move this effort btw?

**Comment by billksun** [comment]
> Hmm how else can we move this effort forward?

I would get on the main Nix/NixOS Matrix channel and see if you can get in touch with a mod there to create an official channel for COSMIC DE.

**Comment by Nyabinary** [comment]
> > Hmm how else can we move this effort forward?
> 
> I would get on the main Nix/NixOS Matrix channel and see if you can get in touch with a mod there to create an official channel for COSMIC DE.
I think this is the official channel for Matrix suggestions in the NixOS space.
https://matrix.to/#/#matrix-discussion:nixos.org

**Comment by Nyabinary** [comment]
Were an official Nix Matrix room now!
https://matrix.to/#/#cosmic:nixos.org

**Comment by foo-dogsquared** [comment]
Anyone interested in adapting https://github.com/NixOS/nixpkgs/pull/250891? I'm also going to create a home-manager version of it which should be the same module except with a different plugin path.

**Comment by ahoneybun** [comment]
> Anyone interested in adapting #250891? I'm also going to create a home-manager version of it which should be the same module except with a different plugin path.

That should be from the COSMIC which is based on GNOME, this is all about the COSMIC DE written in Rust.

**Comment by kirawi** [comment]
I think #199563 is the same thing as this issue

**Comment by Mawfyy** [comment]
There's some discord server or something?

**Comment by ahoneybun** [comment]
> There's some discord server or something?

There is this:

https://github.com/NixOS/nixpkgs/issues/259641#issuecomment-1783830838

**Comment by Mawfyy** [comment]
> > There's some discord server or something?
> 
> There is this:
> 
> [#259641 (comment)](https://github.com/NixOS/nixpkgs/issues/259641#issuecomment-1783830838)

Thanks!!

**Comment by nonetrix** [comment]
I think cosmic terminal is using software rendering in the Nix package, does anyone else have this? Should I make a issue, it doesn't appear in nvtop and uses a lot of CPU doing basic things

**Comment by RuRo** [comment]
It seems that currently all the items in the list are already marked as completed. I was wondering if we are now waiting for upstream to release a stable (or public beta?) version, or if there are still some things missing in nixpkgs?

**Comment by dunxen** [comment]
> It seems that currently all the items in the list are already marked as completed. I was wondering if we are now waiting for upstream to release a stable (or public beta?) version, or if there are still some things missing in nixpkgs?

I believe the NixOS module is not ready. Also, best case scenario is an official Alpha by March: https://blog.system76.com/post/cosmic-the-road-to-alpha

**Comment by RAVENz46** [comment]
I think [cosmic-store](https://github.com/pop-os/cosmic-store), [cosmic-player](https://github.com/pop-os/cosmic-player) and cosmic-greeter module like `services.xserver.displayManager.cosmic-greeter` also missing.
Anyway, wait patiently for the alpha!

**Comment by nyabinary** [comment]
If anyone wants to contribute to the cosmic-greeter module, I'm open to reviewing it and giving it a green light :3

**Comment by Aleksanaa** [comment]
Why cosmic isn't packaged in a scope/set like other desktop environments?

**Comment by a-kenji** [comment]
What advantage would that give us as opposed to the current `by-name` package definitions @Aleksanaa ?

**Comment by Aleksanaa** [comment]
> What advantage would that give us as opposed to the current `by-name` package definitions @Aleksanaa ?

I don't know, and I'm not recommending you do that. But are there any standards in this regard?

**Comment by ahoneybun** [comment]
> Am I expeced to simply add all the Cosmic packages to my configuration, and it will simply offer itself in the login manager?
> 
> Is there a step by step guide, how to try it?

There is not a step by step yet, I think it will be similar to how GNOME and such is but I'm not sure yet.

**Comment by yochananmarqos** [comment]
@ShalokShalom It includes a session that will indeed at it to your Display Manager. 

However, if you have to ask, then you should probably wait. This is neither the place or the time to ask others to hold your hand. :wink: 

This is a development thread, please refrain from off-topic questions and discussion.

**Comment by ShalokShalom** [comment]
Yeah, I agree. And testing is probably a part of the development, and when some people are capable to test, but need step by step instructions to get the environment, it might be beneficial to provide that. 

And considering the emoji response, I might be in good company with that opinion. 👍🏻

**Comment by Aleksanaa** [comment]
> This is a development thread, please refrain from off-topic questions and discussion.

Testing methods are also a crucial part of the development thread. Without proper testing, we all don't know if we've done things right. Please be kind to others even if you are not going to offer any help.

**Comment by lilyinstarlight** [comment]
> A few days ago, cosmic-store became available. Notice that it is not yet ready for use. It seems like it needs some patches to make it works on NixOS?

Flatpaks work already iirc, and we're probably not going to enable packagekit support given how not-well it works and that it tends to break stuff

**Comment by lilyinstarlight** [comment]
> Hi, can we build this now even if it unstable? If that is the case are there instructions on how to this? Thanks

Currently you can test from <https://github.com/lilyinstarlight/nixos-cosmic>, which updates daily, provides a binary cache, and will be merged back to nixpkgs once we start getting tagged releases (e.g. cosmic alpha release)

**Comment by Titaniumtown** [comment]
@lilyinstarlight you're amazing! Thank you for the hard work.

**Comment by kacperwyczawski** [comment]
There is a new blog post from System76 regarding COSMIC:

https://blog.system76.com/post/hammering-out-cosmic-features

**Comment by maupind** [comment]
Hi all,

I have been using NixOS for a couple years now, but only have very minor contributions. I was wondering how I can assist in getting COSMIC working on NixOS. Thanks in advance and apologies if this is the wrong thread to be asking

**Comment by dunxen** [comment]
> I was wondering how I can assist in getting COSMIC working on NixOS.

Hey! I think a great way is to start testing it out if you haven't already via here https://github.com/NixOS/nixpkgs/issues/259641#issuecomment-2025067539

**Comment by RAVENz46** [comment]
A new package, [cosmic-reader](https://github.com/pop-os/cosmic-reader) is coming!

**Comment by MrGamerDoesGames** [comment]
I also want to test for COSMIC, I will try daily driving and see if there are any issues with the nix package or any of that such.

**Comment by amozeo** [comment]
Development of modules is happening in https://github.com/lilyinstarlight/nixos-cosmic

> I also want to test for COSMIC, I will try daily driving and see if there are any issues with the nix package or any of that such.
@MrGamerDoesGames

**Comment by MrGamerDoesGames** [comment]
> Development of modules is happening in https://github.com/lilyinstarlight/nixos-cosmic
> 
> > I also want to test for COSMIC, I will try daily driving and see if there are any issues with the nix package or any of that such.
> > @MrGamerDoesGames

Hello! I'm trying to set up the flake. This is my first time working with flakes. I copied the flake text into flake.nix, saved it, (in my /etc/nixos directory), and ran 'sudo nixos rebuild --switch'. It downloaded successfully, and I put the "desktopManager" thing in my nix config, but now when I try to run the command I get this error message, and it fails:

`error: flake 'path:/etc/nixos' does not provide attribute 'packages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild', 'legacyPackages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild' or 'nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild'`

nix flake update in the directory doesn't work either. I'm sorry if I'm just making some noob mistake or something. I can send screenshots or logs if needed.

**Comment by krovuxdev** [comment]
> > Development of modules is happening in https://github.com/lilyinstarlight/nixos-cosmic
> > > I also want to test for COSMIC, I will try daily driving and see if there are any issues with the nix package or any of that such.
> > > @MrGamerDoesGames
> 
> Hello! I'm trying to set up the flake. This is my first time working with flakes. I copied the flake text into flake.nix, saved it, (in my /etc/nixos directory), and ran 'sudo nixos rebuild --switch'. It downloaded successfully, and I put the "desktopManager" thing in my nix config, but now when I try to run the command I get this error message, and it fails:
> 
> `error: flake 'path:/etc/nixos' does not provide attribute 'packages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild', 'legacyPackages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild' or 'nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild'`
> 
> nix flake update in the directory doesn't work either. I'm sorry if I'm just making some noob mistake or something. I can send screenshots or logs if needed.

this is use command `sudo nixos-rebuild switch .` or  `sudo nixos-rebuild switch --flake .`  for  build you system, because your command `sudo nixos build --switch` is to build binary

**Comment by MrGamerDoesGames** [comment]
> > > Development of modules is happening in https://github.com/lilyinstarlight/nixos-cosmic
> > > > I also want to test for COSMIC, I will try daily driving and see if there are any issues with the nix package or any of that such.
> > > > @MrGamerDoesGames
> > 
> > 
> > Hello! I'm trying to set up the flake. This is my first time working with flakes. I copied the flake text into flake.nix, saved it, (in my /etc/nixos directory), and ran 'sudo nixos rebuild --switch'. It downloaded successfully, and I put the "desktopManager" thing in my nix config, but now when I try to run the command I get this error message, and it fails:
> > `error: flake 'path:/etc/nixos' does not provide attribute 'packages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild', 'legacyPackages.x86_64-linux.nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild' or 'nixosConfigurations."NixOSurface".config.system.build.nixos-rebuild'`
> > nix flake update in the directory doesn't work either. I'm sorry if I'm just making some noob mistake or something. I can send screenshots or logs if needed.
> 
> this is use command `sudo nixos-rebuild switch .` or `sudo nixos-rebuild switch --flake .` for build you system, because your command `sudo nixos build --switch` is to build binary

Thank you, I made a typo in the first issue. Running `sudo nixos-rebuild switch --flake` let it build with no issues, but now I am having trouble making the entry appear in my login manager (SDDM). If there is something I can do to add it there, or to just invoke the DE from a tty, would be appreciated.

**Comment by mxkrsv** [comment]
> I copied the flake text into flake.nix, saved it, (in my /etc/nixos directory), and ran 'sudo nixos rebuild --switch'.

That's not how flakes work. You've just replaced your system with nixos-cosmic flake. You have to add nixos-cosmic to your own system flake's inputs instead.

**Comment by ahoneybun** [comment]
Let's move any troubleshooting of nixos-cosmic into that repo rather then this issue for tracking the packaging of COSMIC on NixOS.

**Comment by MrGamerDoesGames** [comment]
Update: I'm really sorry to all the maintainers, this was not the right place for this. I was being stupid, and I had absolutely zero knowledge of flakes at all. I've educated myself on how they work and I'll try it again tomorrow. I'll delete all my previous comments as they were not on topic. Again, sorry for the waste of time and resources.

**Comment by ahoneybun** [comment]
> Update: I'm really sorry to all the maintainers, this was not the right place for this. I was being stupid, and I had absolutely zero knowledge of flakes at all. I've educated myself on how they work and I'll try it again tomorrow. I'll delete all my previous comments as they were not on topic. Again, sorry for the waste of time and resources.

No worries as everyone is learning and I know flakes have a higher then normal learning curve.

**Comment by Thatoo** [comment]
Will it be possible to install/test Cosmic on stable 24.05 in few days or it will still be usable only on unstable?

**Comment by drakon64** [comment]
> Will it be possible to install/test Cosmic on stable 24.05 in few days or it will still be usable only on unstable?

I tried using the `nixos-cosmic` flake against the `nixos-24.05` channel. I don't remember the exact error but I'm pretty sure it failed to evaluate.

**Comment by drakon64** [comment]
https://github.com/lilyinstarlight/nixos-cosmic works against NixOS 24.05 now

**Comment by ahoneybun** [comment]
@nyabinary cosmic-store is in nixpkgs master now I believe.

**Comment by shackra** [comment]
> @nyabinary cosmic-store is in nixpkgs master now I believe.

yes: https://search.nixos.org/packages?channel=unstable&show=cosmic-store&from=0&size=50&sort=relevance&type=packages&query=cosmic-store

**Comment by Mic92** [comment]
Btw. there is also a NixOS wiki article about the topic: https://wiki.nixos.org/wiki/COSMIC
I linked the flake of lilyin, but not sure if it is missing anything else.

**Comment by jh-devv** [comment]
Happy to assist if something needs spare hands!

**Comment by jh-devv** [comment]
A new package, [cosmic-player](https://github.com/pop-os/cosmic-player) has now a pull request https://github.com/NixOS/nixpkgs/pull/330167!

**Comment by ChocolateLoverRaj** [comment]
How to set COSMIC greeter as the display manager?

**Comment by nonetrix** [comment]
Intrested how to do that too if it works well with Hyprland, could see myself using that before Cosmic comes out doesn't have to do much to work

**Comment by jh-devv** [comment]
[nixos-cosmic](https://github.com/lilyinstarlight/nixos-cosmic) by @lilyinstarlight has an example implementation [here](https://github.com/lilyinstarlight/nixos-cosmic/blob/main/nixos/cosmic/module.nix). Hope it helps!

**Comment by bjsemrad** [comment]
> Intrested how to do that too if it works well with Hyprland, could see myself using that before Cosmic comes out doesn't have to do much to work

I have Hyprland, Cosmic and Gnome on a single machine. I use GDM for greeter at the moment.

**Comment by Thatoo** [comment]
You can add [Quick Web Apps](https://github.com/elevenhsoft/WebApps?tab=readme-ov-file#readme) tracked here : https://github.com/NixOS/nixpkgs/pull/322226

**Comment by ahoneybun** [comment]
> You can add [Quick Web Apps](https://github.com/elevenhsoft/WebApps?tab=readme-ov-file#readme) tracked here : #322226

That is not an official app part of COSMIC DE though.

**Comment by nonetrix** [comment]
Yes, make a separate issue for that if you want it packaged. Seems interesting, hope more unofficial libcosmic apps come out eventually

**Comment by Pandapip1** [comment]
Consider adding https://github.com/NixOS/nixpkgs/pull/348146?

**Comment by nyabinary** [comment]
> @nyabinary can you add cosmic-wallpapers and remove the calculator? That is not an official part of COSMIC but a 3rd party application.

Done :3

**Comment by jh-devv** [comment]
Is there something else that need's work? @ahoneybun had some updates for #330167 but other than that, I really don't know?
I have bunch of free time this week and after the next week, the couple of weeks. So, I could look into something, maybe even some home manager stuff?

**Comment by ahoneybun** [comment]
@jh-devv I have PR's for COSMIC Settings and COSMIC Files that could use help on if you have time.

**Comment by FliegendeWurst** [comment]
PR for cosmic-calculator: #340725

**Comment by jh-devv** [comment]
@nyabinary Hey, we made a pull request for `cosmic-player` due to some fork whoopsies at my end. Could you update the description to change the PR to #350710?

**Comment by HeitorAugustoLN** [comment]
Up-to-date PR for cosmic-calculator: #379412
Also cosmic-player is now merged after #373897

**Comment by HeitorAugustoLN** [comment]
PR for [COSMIC Tweaks](https://github.com/cosmic-utils/tweaks): #379419

**Comment by HeitorAugustoLN** [comment]
New PR for libcosmicAppHook: #380312

**Comment by HeitorAugustoLN** [comment]
cosmic-idle: #380386

**Comment by Thatoo** [comment]
In case it could give the idea to someone able to make it: 

> it would be nice to have a nixos "update" cosmic applet. An applet that regularly check if updates are available, shows a notification to let know when some are available and offers a button to launch upgrade (nixos-rebuild).

**Comment by spaceoden** [comment]
cosmic-ext-tweaks is 3rd party so shouldn't be included , right?

**Comment by HeitorAugustoLN** [comment]
> cosmic-ext-tweaks is 3rd party so shouldn't be included , right?

There could be a separate part for community apps

**Comment by rhoriguchi** [comment]
Is it possible to add an [excludePackages](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=excludePackages) option like other DE have. Only make a small required base none excludable.

**Comment by isabelroses** [comment]
> Is it possible to add an [excludePackages](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=excludePackages) option like other DE have. Only make a small required base none excludable.

This seems to have been dropped from the original modules https://github.com/lilyinstarlight/nixos-cosmic/blob/cd185ca7ed5fc0ad766370abc48294abe36dec52/nixos/cosmic/module.nix#L31

**Comment by bendi-github** [comment]
> > I wanted to try out the new cosmic PR today and noticed that my machine tried to build "cosmic-comp" from source (which failed, because I ran out of memory).
> > 
> > I tested whether the package is in the cache
> > As per https://wiki.nixos.org/wiki/Binary_Cache
> > `curl https://cache.nixos.org/sw0ikccc9mqpnf5givbh9xa3nxs84dyn.narinfo`
> > shows that the package is in the cache.
> > 
> > As far as I can tell this behavior is caused by the override in the [module definition](https://github.com/thefossguy/nixpkgs/blob/64ce11200739d82a67cf73c0a43f2775f15e5809/nixos/modules/services/desktop-managers/cosmic.nix#L49)
> > Is this intended? To me it seems like most of cosmic's users would be using the nixos module and therefore not benefit from the binary cache.
> 
> Is there a reason to disable XWayland in the module?

https://github.com/lilyinstarlight/nixos-cosmic/blob/cd185ca7ed5fc0ad766370abc48294abe36dec52/nixos/cosmic/module.nix#L63

**Comment by drakon64** [comment]
Perhaps the package should have that disabled by default, that way the users of the module can benefit from the binary cache.

I'll raise a PR.

**Comment by nyabinary** [comment]
Why doesn't it cache with XWayland on anyways?

**Comment by drakon64** [comment]
> Why doesn't it cache with XWayland on anyways?

Does Hydra build the package as defined by the module or just as what the package defines with no overrides?

**Comment by FliegendeWurst** [comment]
I think Hydra would build it, if it were used in a NixOS test.

Another solution would be to add something like `cosmic-comp_noXWayland = cosmic-comp.override { useXWayland = false; };` to all-packages.

**Comment by thefossguy** [comment]
This point was discussed in the PR adding the cosmic modules. The main issue is that XWayland, unless specified otherwise, is enabled by default in the pacakge. Then, we go ahead and disable it in the cosmic module to not clutter the PATH. The cosmic-comp package with XWayland disabled doesn't get built by Hydra because we're not consuming that package anywhere (yet). As @FliegendeWurst said, if we had a test, cosmic-comp without XWayland would be consumed and Hydra will build it. But we don't. I'm waiting on #395741 to get merged so that I can push 2 cosmic tests.

IMHO, that package override doesn't make any sense (at least not to me) but that's what was asked to get it merged so I changed it and now we have a cache miss for a package which requires a lot of resources for building locally.

**Comment by thefossguy** [comment]
Opened a PR upstream adding install instructions for NixOS: https://github.com/pop-os/cosmic-epoch/pull/1832

**Comment by thefossguy** [comment]
Is there a way to opt into notifications about build failures from Hydra? I wouldn't mind some noise in my inbox.

**Comment by JohnRTitor** [comment]
> Is there a way to opt into notifications about build failures from Hydra? I wouldn't mind some noise in my inbox.

Not sure, perhaps @mweinelt knows.

**Comment by mweinelt** [comment]
Notifications no, but a per maintainer overview exists for example on zh.fail.

**Comment by thefossguy** [comment]
I agree with @JohnRTitor. Since this variable gets updated post login, it is the desktop environment's job to update it as well. I'll create a PR upstream addressing this.

**Comment by Occhioverde** [comment]
Hi all!

I don't know if this is a problem with my system configuration, but since I activated the Cosmic module my computer started hanging at `A stop job is running for Session x of user rsacchetto (xs / 1min 30)` while shutting down; checking the logs with `journalctl -b -1 -e` yields the following warnings:

```
May 23 00:48:02 Ricky-InfinityBook systemd[1]: session-3.scope: Stopping timed out. Killing.
May 23 00:48:02 Ricky-InfinityBook systemd[1]: session-3.scope: Killing process 2255 (cosmic-settings) with signal SIGKILL.
May 23 00:48:02 Ricky-InfinityBook systemd[1]: session-3.scope: Killing process 2260 (notify-rs inoti) with signal SIGKILL.
May 23 00:48:02 Ricky-InfinityBook systemd[1]: session-3.scope: Killing process 19709 (n/a) with signal SIGKILL.
May 23 00:48:02 Ricky-InfinityBook systemd[1]: session-3.scope: Failed with result 'timeout'.
```

Is there some known problem with these services that I was unable to find, or am I the first one to experience this?

**Comment by michaelBelsanti** [comment]
> Is there some known problem with these services that I was unable to find, or am I the first one to experience this?

Theres an issue for this upstream:
https://github.com/pop-os/cosmic-settings/issues/1154

Someone was able to reproduce it on PopOS as well, although they later said that is stopped occurring.

**Comment by Thatoo** [comment]
What about COSMIC examine https://github.com/cosmic-utils/examine ?

**Comment by Thatoo** [comment]
and COSMIC reader https://github.com/pop-os/cosmic-reader ?

**Comment by Thatoo** [comment]
and COSMIC Clipboard Manager : https://github.com/cosmic-utils/clipboard-manager ?

**Comment by GaetanLepage** [comment]
Feel free to open PRs @Thatoo ;)

**Comment by Thatoo** [comment]
I face an issue, Clicking on URL don't open default browser : 
I have
```
~$ xdg-settings get default-web-browser
librewolf.desktop
```
but when I click on a URL, written in an email in Thunderbird or in a message in Element, nothing happen.

I had a similar issue a year ago using Lily's flake : https://github.com/lilyinstarlight/nixos-cosmic/issues/273
I tried the trick given then but it doesn't work this time.

**Comment by Pandapip1** [comment]
In terms of third party applets, there is https://github.com/technobaboo/caps-lock-indicator; this is on my backlog of things to package, although if anyone else wants to have a go I'd be happy to drop a review.

**Comment by thefossguy** [comment]
Should we namespace the utilities specific to cosmic? Makes it easier to notice that said utility is specifically made for cosmic and may not work in other environments _and_ help discover more utilities by keeping it in a namespace.

**Comment by GaetanLepage** [comment]
> Should we namespace the utilities specific to cosmic? Makes it easier to notice that said utility is specifically made for cosmic and may not work in other environments _and_ help discover more utilities by keeping it in a namespace.

I think that this is a good idea.

**Comment by Pandapip1** [comment]
Most GNOME stuff was actually recently de-namespaced: https://github.com/NixOS/nixpkgs/pull/319659

What's the specific benefit that namespacing the cosmic stuff will bring?

**Comment by michaelBelsanti** [comment]
> That's the same issue as [#259641 (comment)](https://github.com/NixOS/nixpkgs/issues/259641#issuecomment-2898279541)
> 
> it should be fixed by the above solutions, the ideal one being: compositor doing it.

Do you know what causes this issue to be inconsistent across systems? I don't have any issues with links in cosmic.

**Comment by thefossguy** [comment]
I have the same problem with opening links (hadn't noticed it until now).

The following `.desktop` file works:
```
[Desktop Entry]
Version=1.0
Type=Application
Name=GitHub
Comment=Open GitHub in default browser
Exec=xdg-open https://github.com
Icon=web-browser
Terminal=false
Categories=Network;WebBrowser;
StartupNotify=true
```

But if you remove the `xdg-open`, it stops working. I have no idea why updating the `PATH` for the user's systemd session would have any impact on opening links.

**Comment by Thatoo** [comment]
> > I face an issue, Clicking on URL don't open default browser : I have
> > ```
> > ~$ xdg-settings get default-web-browser
> > librewolf.desktop
> > ```
> >   
> > but when I click on a URL, written in an email in Thunderbird or in a message in Element, nothing happen.
> 
> Can you try running 'xdg-open' with a URL and share the output?

I get : 
```
xdg-open - opens a file or URL in the user's preferred application

Synopsis

xdg-open { file | URL }

xdg-open { --help | --manual | --version }

Use 'man xdg-open' or 'xdg-open --manual' for additional info.
```

**Comment by thefossguy** [comment]
You're supposed to run `xdg-open https://gitlab.com` and check if a new tab/window opens visiting GitLab. Mind you, if the browser is open, a new tab will likely open, not a window. So check the existing window(s), if any before concluding the behavior.

**Comment by JohnRTitor** [comment]
> why updating the `PATH` for the user's systemd session would have any impact on opening links

`xdg-open` and xdg portals run as a systemd service, so without importing PATH to systemd, it won't know which apps are available, and which ones should be opened.

**Comment by Thatoo** [comment]
> You're supposed to run `xdg-open https://gitlab.com` and check if a new tab/window opens visiting GitLab. Mind you, if the browser is open, a new tab will likely open, not a window. So check the existing window(s), if any before concluding the behavior.

If I **don't have** in configuration.nix file
`      systemd.user.extraConfig = ''DefaultEnvironment="PATH=/run/wrappers/bin:/etc/profiles/per-user/%u/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin"'';`
Clicking on a link in other app such as Thunderbird, **doesn't open** it in my web browser.
However, launching `xdg-open https://gitlab.com` in terminal does open a tab in the web browser.


If I **do have** in configuration.nix file the line
`      systemd.user.extraConfig = ''DefaultEnvironment="PATH=/run/wrappers/bin:/etc/profiles/per-user/%u/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin"'';`
Clicking on a link in other app such as Thunderbird, **DO open** it in my web browser.
And of course, launching `xdg-open https://gitlab.com` in terminal does open a tab in the web browser too.

**Comment by thefossguy** [comment]
Weird, I can open links from thunderbird without needing that PATH update for systemd. Is your thunderbird binary a Nix binary or a flatpak?

**Comment by Thatoo** [comment]
My thunderbird is a Nix binary, I don't use flatpak. By the way, link aren't open when I click on them form any app, not just Thunderbird (well except within my web browser of course :-)

**Comment by Thatoo** [comment]
Just to be sure, here is what I wrote in my flake.nix file : 
```
inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-25.05";
  };

  outputs = inputs@{ self, nixpkgs, ... }: {
    nixosConfigurations.Laptop = nixpkgs.lib.nixosSystem {
      system = "x86_64-linux";
      specialArgs = { inherit inputs; };
      modules = [
        ./hardware-configuration.nix
        ./configuration.nix
        ./system_packages.nix
        ./serveur_web.nix
      ];
    };
  };
}
```
and here is what I wrote in configuration.nix : 
```
{ config, pkgs, lib, inputs,  ... }:

{
  services = {
    xserver.enable = true;
    # Enable the login manager
    displayManager.cosmic-greeter.enable = true;
    # Enable the COSMIC DE itself
    desktopManager.cosmic.enable = true;
    # Enable XWayland support in COSMIC
    desktopManager.cosmic.xwayland.enable = true;
  };

  environment.sessionVariables.COSMIC_DATA_CONTROL_ENABLED = 1;

  systemd.user.extraConfig = ''DefaultEnvironment="PATH=/run/wrappers/bin:/etc/profiles/per-user/%u/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin"'';
}
```

**Comment by michaelBelsanti** [comment]
https://github.com/pop-os/cosmic-settings/issues/1154

Linking to this issue for visibility if anyone searches Nixpkgs trying to fix it. `cosmic-settings-daemon` will currently hang on shutdown if geoclue2 isn't running. You can avoid this by setting `services.geoclue2.enable = true;`

**Comment by Thatoo** [comment]
> I face a weird issue : in some circumstances that I don't know yet (sounds random so far), after launching an app from the panel, it makes all app launchers disappear from the panel. I need to relaunch the panel to get apps back.

Well, now I can say this happen when I'm using my laptop screen only. When i work with my external screens, then I don't face this issue at all. However, I don't know yet when it happens when I'm using the small screen of my laptop. For the record, I didn't face this issue when using Lily's flake.

**Comment by ahoneybun** [comment]
Fixed or at least a workaround in this PR:

https://github.com/NixOS/nixpkgs/pull/412260

**Comment by Thatoo** [comment]
The notification center applet doesn't display/work so far.

**Comment by Thatoo** [comment]
I face this issue https://github.com/pop-os/cosmic-epoch/issues/1746 that is supposed to be fixed upstream and was fixed in last version of Lily's flake. Is it normal and only a question of time to be available in 25.05?

**Comment by thefossguy** [comment]
Sorry for the late reply. The reporter mentions that the issue is fixed in version `1.0.0~alpha.7^git20250502.1d53973` (https://github.com/pop-os/cosmic-applibrary/issues/219#issuecomment-2848593085) and it refers to this commit: https://github.com/pop-os/cosmic-applibrary/commit/1d5397307e1247662465cb40e67173b85b10153e. This commit was made _after_ the Alpha 7 got tagged and therefore it isn't fixed (in NixOS), _yet_. It will be included in the first beta, whenever it is available.

That said, _I_ don't intend to backport nor support any changes to `nixos-${stable}` for the time being. `nixos-unstable` is where it's at for developing COSMIC for the moment.

**Comment by Thatoo** [comment]
Ok, got it.

> That said, _I_ don't intend to backport nor support any changes to `nixos-${stable}` for the time being. `nixos-unstable` is where it's at for developing COSMIC for the moment.

That is sad. I'll stop reporting bug then as my config will soon be outdated.

**Comment by thefossguy** [comment]
Doesn't matter. If you have a bug, **please report it**. While the bug that you reported isn't caused by us, there might be a bug caused by us in the future and I don't want to miss out on that.

**Comment by Thatoo** [comment]
> The notification center applet doesn't display/work so far.

After last time I did (yesterday)
```
sudo nix flake update
sudo nixos-rebuild switch --show-trace --flake .#Desktop
```
, the notification center applet has appeared.

**Comment by Thatoo** [comment]
When I shutdown my computer, recently, I have to wait 1min 30 sec that a job end.

**Comment by Thatoo** [comment]
After updating this evening, after a reboot, I face an issue with Electron apps. When I launch it, I get a warning window saying something like "your system has an unsupported key ring, the database cannot be opened." And it points to this page : https;//www.electronjs.org/docs/latest/api/safe-storage#safestoragegetselectedstoragebackend-linux

![Image](https://github.com/user-attachments/assets/c4dc17cc-9a26-4c5b-bfcc-b75bb4841921)

I reboot and switch back to my previous nixos build then.

**Comment by michaelBelsanti** [comment]
> When I shutdown my computer, recently, I have to wait 1min 30 sec that a job end.

Probably this issue https://github.com/pop-os/cosmic-settings/issues/1154

Enabling geoclue2 should fix it https://github.com/NixOS/nixpkgs/pull/412260/files

**Comment by thefossguy** [comment]
#412260 should've fixed it

**Comment by Thatoo** [comment]
It either not yet on 25.04 stable or it's un the last build (yesterday evening update) but then un this update something create an issue with Element electron app.

**Comment by giomf** [comment]
>  #412260 should've fixed it

I'm on unstable and this has not fixed the issue for me.

**Comment by michaelBelsanti** [comment]
That PR wasnt backported to stable, you can enable the same options in your own config to fix the issue

**Comment by Thatoo** [comment]
> When I shutdown my computer, recently, I have to wait 1min 30 sec that a job end.

Enabling geoclue2 did indeed solve the issue.

**Comment by Thatoo** [comment]
> After updating this evening, after a reboot, I face an issue with Electron apps. When I launch it, I get a warning window saying something like "your system has an unsupported key ring, the database cannot be opened." And it points to this page : https;//www.electronjs.org/docs/latest/api/safe-storage#safestoragegetselectedstoragebackend-linux
> 
> ![Image](https://github.com/user-attachments/assets/c4dc17cc-9a26-4c5b-bfcc-b75bb4841921)
> 
> I reboot and switch back to my previous nixos build then.

Indeed, to solve the issue, I had to add --password-store=gnome-libsecret to the Exec line of my .desktop file.
Not very handy though...
Hopefully, this issue will be solved soon : https://github.com/electron/electron/issues/47436

**Comment by Thatoo** [comment]
Someone address a solution to this issue "add language settings unavailable" here https://github.com/lilyinstarlight/nixos-cosmic/issues/566#issuecomment-3008076200 . I didn't test it yet (not sure how) but I think I better share it here.

I tried to add this but it doesn't build : 
```
nixpkgs.overlays = [ ( final: prev:
  prev.cosmic-settings.overrideAttrs (oldAttrs: rec {
    src = prev.fetchFromGitHub {
      owner = "kajusnau";
      repo = "cosmic-settings";
      rev = "locale-rs-patch";
      hash = "sha256-8VO17z2nd1PGe1rWyAky06HyLRqrV8DZCEvZ1AUICoI=";
    };
    cargoDeps = prev.rustPlatform.fetchCargoVendor {
      inherit src;
      hash = "sha256-CpBa2NLirHMcSMO0l8mxD+d6JPP6qkbtdYYrUqgWBuE=";
    };
  })
) ];
```
I get errors.

**Comment by thefossguy** [comment]
> Someone address a solution to this issue "add language settings unavailable" here [lilyinstarlight/nixos-cosmic#566 (comment)](https://github.com/lilyinstarlight/nixos-cosmic/issues/566#issuecomment-3008076200) . I didn't test it yet (not sure how) but I think I better share it here.
> 
> I tried to add this but it doesn't build :
> 
> ```
> nixpkgs.overlays = [ ( final: prev:
>   prev.cosmic-settings.overrideAttrs (oldAttrs: rec {
>     src = prev.fetchFromGitHub {
>       owner = "kajusnau";
>       repo = "cosmic-settings";
>       rev = "locale-rs-patch";
>       hash = "sha256-8VO17z2nd1PGe1rWyAky06HyLRqrV8DZCEvZ1AUICoI=";
>     };
>     cargoDeps = prev.rustPlatform.fetchCargoVendor {
>       inherit src;
>       hash = "sha256-CpBa2NLirHMcSMO0l8mxD+d6JPP6qkbtdYYrUqgWBuE=";
>     };
>   })
> ) ];
> ```
> 
> I get errors.

Since it is an update to a rust library, its best to address this issue in the next release instead of carrying a Cargo.lock file in nixpkgs.

**Comment by kajusnau** [comment]
> Someone address a solution to this issue "add language settings unavailable" here [lilyinstarlight/nixos-cosmic#566 (comment)](https://github.com/lilyinstarlight/nixos-cosmic/issues/566#issuecomment-3008076200) . I didn't test it yet (not sure how) but I think I better share it here.
> 
> I tried to add this but it doesn't build :
> 
> ```
> nixpkgs.overlays = [ ( final: prev:
>   prev.cosmic-settings.overrideAttrs (oldAttrs: rec {
>     src = prev.fetchFromGitHub {
> ...
> I get errors.

The fix has been merged in the upstream `locales-rs` and should be available in the next release of Cosmic.
If you still wish to test it before the next release, you can:

1. Take the above mentioned overlay and replace the hash values with empty strings: `hash = "";`
2. Run `nix build` , wait for it to fail, and apply the new hash value from the error output, e.g.: `got:    sha256-zTUv/duWE6HYraK9874u1xd8TA7POKim0RCzaLtryLk=`
3. Perform step 2 twice, as both hash values must be replaced.

Like @thefossguy mentioned, this should **not** be done in nixpkgs, only in your local config if you need this functionality urgently before the next release.

**Comment by Pandapip1** [comment]
We could ask upstream COSMIC to update their `Cargo.lock` and then backport that commit, assuming `locales-rs` has a new release with that fix. I wouldn't be opposed to that. Definitely -1 to pointing `src` to anything other than the official repository.

**Comment by Pandapip1** [comment]
It also looks like `cosmic-screenshot` right now just... doesn't work. The following messages appeared in stderr after I ctrl-C'd it after waiting for ~10 mins for anything to happen:

```
~ via  v3.11.12 took 29s
❯ cosmic-screenshot

thread 'main' panicked at src/main.rs:70:10:
failed to send screenshot request: Zbus(MethodError(OwnedErrorName(ErrorName(Str(Owned("org.freedesktop.DBus.Error.TimedOut")))), Some("Failed to activate service 'org.freedesktop.portal.Desktop': timed out (service_start_timeout=120000ms)"), Msg { type: Error, sender: UniqueName(Str(Borrowed("org.freedesktop.DBus"))), reply-serial: 17, body: Signature("s") }))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
```

**Comment by thefossguy** [comment]
@Pandapip1 does cosmic-screenshot work again now that #419948 is available nixos-unstable?

**Comment by Pandapip1** [comment]
Sorry for the late reply--yes.

**Comment by kajusnau** [comment]
> > [cosmic-reader](https://github.com/pop-os/cosmic-reader)
> 
> Any update on its integration within NixOS COSMIC Desktop Environment?
> 
> What about external applet such as
> 
> * cosmic-ext-applet-clipboard-manager
> * cosmic-ext-applet-external-monitor-brightness
>   ?

None of those tools are in `nixpkgs` yet

**Comment by candsastle** [comment]
Are there plans to introduce additional NixOS options for COSMIC beyond enabling it? For example, selectively turning certain COSMIC apps on/off or configuring specific apps? I know there's already the [cosmic-manager ](https://github.com/HeitorAugustoLN/cosmic-manager)flake, but it seems like it errors out for me and is (for the time being) abandoned.

**Comment by ahoneybun** [comment]
I'm pretty sure you can already select to remove certain apps, some are required for the desktop to work though. > Are there plans to introduce additional NixOS options for COSMIC beyond enabling it? For example, selectively turning certain COSMIC apps on/off or configuring specific apps? I know there's already the [cosmic-manager ](https://github.com/HeitorAugustoLN/cosmic-manager)flake, but it seems like it errors out for me and is (for the time being) abandoned.

**Comment by SteamRex25** [comment]
> > I just hope we get some of the bugfixes that have been released as part of more recent released of alpha 7
> 
> Alpha 7 came out in April and that is in NixOS 25.05 currently, for the very latest NixOS unstable would be the place.

Im on unstable but the bugs I'm seeing are bugs from versions of various cosmic functions that are a bit old, I've heard about the nightly flake but idk if I should use it to get newer packages

**Comment by HeitorAugustoLN** [comment]
> > > I'm pretty sure you can remove certain apps already, some are needed for the desktop to work through.
> > 
> > 
> > So I just checked and indeed there is a `environment.cosmic.excludePackages` option, but only for Unstable NixOS. Which is fine, since I can wait until 25.11 becomes stable. I guess I was more leaning more towards configuring system settings and application settings, but I guess some of that could be more in the realm of Home Manager options.
> 
> I suspect cosmic-manager will be more usable once COSMIC has it's first stable release and [@HeitorAugustoLN](https://github.com/HeitorAugustoLN) has more time with it as he is part of the COSMIC team handling the packaging and updates.

`cosmic-manager` is fully usable, though it still needs some documentation. I'm currently upstreaming some of its components to `nixpkgs` (#440544) to enable system-wide COSMIC configurations (#440803).

I also plan to upstream the core modules to `home-manager`. However, the high-level, type-safe modules for specific applications (like `cosmic-files`) or COSMIC components (like `appearance`) will remain in `cosmic-manager`. Because it is probably out-of-scope for home-manager.

**Comment by Pheoxy** [comment]
> > > I just hope we get some of the bugfixes that have been released as part of more recent released of alpha 7
> > 
> > Alpha 7 came out in April and that is in NixOS 25.05 currently, for the very latest NixOS unstable would be the place.
> 
> Im on unstable but the bugs I'm seeing are bugs from versions of various cosmic functions that are a bit old, I've heard about the nightly flake but idk if I should use it to get newer packages

Same, I'm hoping they do another release soon which by their updates may be a beta instead of alpha by the end of this month.

**Comment by candsastle** [comment]
> I also plan to upstream the core modules to `home-manager`. However, the high-level, type-safe modules for specific applications (like `cosmic-files`) or COSMIC components (like `appearance`) will remain in `cosmic-manager`. Because it is probably out-of-scope for home-manager.

Have you thought about upstreaming some of those options to the NixOS Cosmic module?

**Comment by SteamRex25** [comment]
> We will have tags for distros like nixos to use in advance of that 🤞

so will we be getting a new release soon?

**Comment by thefossguy** [comment]
> The Cosmic Desktop beta is out, could we get an update on `services.desktopManager.cosmic.enable` ?

I don't follow. What kind of update are you hoping for/expecting?

**Comment by snoweuph** [comment]
> > The Cosmic Desktop beta is out, could we get an update on `services.desktopManager.cosmic.enable` ?
> 
> COSMIC was updated in [#440950](https://github.com/NixOS/nixpkgs/pull/440950), expect it to be on `nixos-unstable` in the next few days

thanks :D 

can't wait to try out the latest versions

**Comment by Pandapip1** [comment]
Looks like it's landed in unstable!

https://nixpkgs-tracker.ocfox.me/?pr=440950

**Comment by esto20** [comment]
Having an issue with cosmic-comp after updating and trying to rebuild on unstable. Can't reach log in screen / cosmic greeter.

journalctl --boot=-1 -p 3 -xb:

Nov 26 07:53:54 nixos .cosmic-comp-wrapped[1432]: [EGL] 0x3008 (BAD_DISPLAY) eglCreateSync: _eglCreateSync
Nov 26 07:53:49 nixos .cosmic-greeter-wrapped[1500]: failed to load user data from daemon: org.freedesktop.DBus.Error.NoReply


Using thinkpad t480.

**Comment by Thatoo** [comment]
I've updated to NIxOS 25.11. I was hoping that cosmic beta would solve my issue but it didn't.
I have nextcloud sync desktop app install. sync is working but I can't open the main window.
I open an issue on cosmic repo (https://github.com/pop-os/cosmic-epoch/issues/2571) but I wanted to know if others are experimenting the same issue using NixOS 25.11?

**Comment by Jingles52** [comment]
yeah, that's what i see too

**Comment by kirillrdy** [comment]
@snoweuph unstable is basically master branch, but at point in time were most builds cached.

there some tools to track progress of PR though branches eg https://nixpkgs-tracker.ocfox.me/?pr=469745

**Comment by BerriJ** [comment]
> Thank, what branch is related to NixOS stable 25.11 then ?

Thats the nixos-25.11 branch. 
Cosmic 1.0.0 will not automatically reach this channel.
For this, a so called Back port is needed. It's planned after we had the 1.0.0 for some time in nixos-unstable. So maybe in 2 weeks or so.

**Comment by snoweuph** [comment]
> There are other changes that are blocked (iirc, failing tests). The build is RAM intensive, so I'd say limit the number of builds and consider using the Garnix cache since some of it is already built.

I don't have the biggest config so I gonna go without any new caches I don't know so far. thanks for the Infos though, 

garnix certainly looks interesting
> DE->EN: noting certainly looks interesting
 
when I have the mental capacity, i will contently look into (during writing this I started the rebuild and it only took a few minutes so I'm fine)

I guess as soon as `nixos-unstable` is build again i can just switch back easly. :D

**Comment by thefossguy** [comment]
Can we get a GitHub project under nixpkgs for COSMIC to better track our "TODOs?" I can't do that because I don't have commit access to nixpkgs.

**Comment by GaetanLepage** [comment]
> Can we get a GitHub project under nixpkgs for COSMIC to better track our "TODOs?" I can't do that because I don't have commit access to nixpkgs.

Good idea. Done: https://github.com/orgs/NixOS/projects/112/

**Comment by SpiderUnderUrBed** [comment]
Dont know if this is the place to mention this, but is anyone else getting this issue with cosmic screenshots?
```
           PID: 77068 (cosmic-screensh)
           UID: 1000 (spiderunderurbed)
           GID: 100 (users)
        Signal: 6 (ABRT)
     Timestamp: Fri 2026-04-17 15:36:24 NZST (7s ago)
  Command Line: cosmic-screenshot
    Executable: /nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot
 Control Group: /user.slice/user-1000.slice/session-12.scope
          Unit: session-12.scope
         Slice: user-1000.slice
       Session: 12
     Owner UID: 1000 (spiderunderurbed)
       Boot ID: ab8c00862ce94edfa0e97bf3882979d0
    Machine ID: 511b644bc6e14c588b91b62e176ca0ac
      Hostname: daspidercave
       Storage: /var/lib/systemd/coredump/core.cosmic-screensh.1000.ab8c00862ce94edfa0e97bf3882979d0.77068.1776396984000000.zst (present)
  Size on Disk: 50.2K
       Message: Process 77068 (cosmic-screensh) of user 1000 dumped core.
                
                Module /nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot without build-id.
                Module libgcc_s.so.1 without build-id.
                Stack trace of thread 77068:
                #0  0x00007f138469fdcc __pthread_kill_implementation (libc.so.6 + 0x9fdcc)
                #1  0x00007f138464265e raise (libc.so.6 + 0x4265e)
                #2  0x00007f1384629350 abort (libc.so.6 + 0x29350)
                #3  0x0000558702fe6a7a _ZN3std3sys3pal4unix14abort_internal17h1252393c5529ccfcE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x20a7a)
                #4  0x0000558702fe6b39 _ZN3std7process5abort17h7df0c7b9857474a6E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x20b39)
                #5  0x0000558702fe6eb9 _RNvCshu4I594XYYe_7___rustc12___rust_abort (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x20eb9)
                #6  0x0000558702fe1839 _RNvCshu4I594XYYe_7___rustc18___rust_start_panic (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x1b839)
                #7  0x0000558702fe6d79 _RNvCshu4I594XYYe_7___rustc10rust_panic (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x20d79)
                #8  0x0000558703107207 _ZN3std9panicking15panic_with_hook17hb09a36a2026ab877E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x141207)
                #9  0x0000558703107c77 _ZN3std9panicking13panic_handler28_$u7b$$u7b$closure$u7d$$u7d$17ha4299fe6dc67fbbcE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x141c77)
                #10 0x0000558703107c29 _ZN3std3sys9backtrace26__rust_end_short_backtrace17hafec91f98267cfd4E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x141c29)
                #11 0x0000558703107c1c _RNvCshu4I594XYYe_7___rustc17rust_begin_unwind (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x141c1c)
                #12 0x0000558702fde3eb _ZN4core9panicking9panic_fmt17h41b75fcbafe65d21E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x183eb)
                #13 0x0000558702fde2c1 _ZN4core6result13unwrap_failed17hec126d4a1961cb02E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x182c1)
                #14 0x000055870300ba38 _ZN17cosmic_screenshot4main28_$u7b$$u7b$closure$u7d$$u7d$17h3cc1b1df54c2923eE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x45a38)
                #15 0x00005587030025c2 _ZN17cosmic_screenshot4main17hc60057beb2cd642aE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x3c5c2)
                #16 0x000055870300e8b6 _ZN3std3sys9backtrace28__rust_begin_short_backtrace17h03b0f8b1f5e884eeE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x488b6)
                #17 0x0000558703072236 main (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0xac236)
                #18 0x00007f138462b285 __libc_start_call_main (libc.so.6 + 0x2b285)
                #19 0x00007f138462b338 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2b338)
                #20 0x0000558702feeb75 _start (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x28b75)
                
                Stack trace of thread 77069:
                #0  0x00007f138472320d syscall (libc.so.6 + 0x12320d)
                #1  0x00005587030ed09b _ZN3std3sys4sync7condvar5futex7Condvar12wait_timeout17h1126b1eebfda20f9E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x12709b)
                #2  0x00005587031231c2 _ZN3std3sys9backtrace28__rust_begin_short_backtrace17h5223c618104fda6dE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x15d1c2)
                #3  0x0000558703125924 _ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h0d3e7f27ca123e5dE (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x15f924)
                #4  0x000055870311018b _ZN3std3sys6thread4unix6Thread3new12thread_start17h58b7137495120f59E (/nix/store/q9a6n2zbz5n25iw235w1xwwlf8v7c55m-cosmic-screenshot-1.0.8/bin/cosmic-screenshot + 0x14a18b)
                #5  0x00007f138469dd53 start_thread (libc.so.6 + 0x9dd53)
                #6  0x00007f138472563c __clone3 (libc.so.6 + 0x12563c)
                ELF object binary architecture: AMD x86-64
```
If for the future i should open separate issues for problems i find with cosmic, please tell me.

**Comment by thefossguy** [comment]
Yes, please create a separate issue about this, @SpiderUnderUrBed. Please also add the COSMIC label to your issue while tagging `NixOS/COSMIC`.
